modules/packages/Sort.chpl

```diff
@@ -591,55 +591,65 @@ proc binaryInsertionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparato
  */
 proc mergeSort(Data: [?Dom] ?eltType, minlen=16, comparator:?rec=defaultComparator) {
   chpl_check_comparator(comparator, eltType);
-  _MergeSort(Data, minlen, comparator);
+  _MergeSort(Data, Dom.low, Dom.high+1, minlen, comparator);
 }
 
-private proc _MergeSort(Data: [?Dom], minlen=16, comparator:?rec=defaultComparator)
+private proc _MergeSort(Data: [?Dom], lo:int, hi:int, minlen=16, comparator:?rec=defaultComparator)
   where Dom.rank == 1 {
-  const lo = Dom.dim(1).low;
-  const hi = Dom.dim(1).high;
-  if hi-lo < minlen {
-    insertionSort(Data, comparator);
+  if (hi-lo < minlen) {
+    var sorted = Data[lo..(hi-1)];
+    insertionSort(sorted, comparator=comparator);
+    for (i) in (lo..(hi-1)) {
+      Data(i) = sorted(i);
+    }
+    return;
+  }
+  // Base cases in case minlen is too low
```

Comment:
I think it'd be better to just use `var localMinlen = if minlen < 2 then 2 else minlen;` and then `if (hi-lo < localMinlen)` above. I don't think we need these base cases in general & they'll increase the code complexity & reduce optimization when we plan to use insertionSort/shellSort for the base case anyway.